h2/src/main/org/h2/api/UserToRolesMapper.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2004-2018 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (http://h2database.com/html/license.html).
+ * Initial Developer: Alessandro Ventura
+ */
+package org.h2.api;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.h2.security.auth.AuthenticationException;
+import org.h2.security.auth.AuthenticationInfo;
+import org.h2.security.auth.Configurable;
+
+/**
+ * A class that implement this interface can be used during
+ * authentication to map external users to database roles.
+ * It is used by DefaultAuthenticator
+ */
+public interface UserToRolesMapper extends Configurable {
```

Comment:
this class looks like it doesn't need to be in the api package

Reply:
It has been put inside org.h2.security.auth package because In my mind this interface can be offered to developers to support their custom need